clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -562,12 +570,52 @@ test_regions_start()
     assert(ordinal == 6);
 }
 
+static void
+test_regions_too_far()
+{
+    std::cerr << "\n----------------\nTesting region going too far\n";
+    std::vector<trace_entry_t> memrefs = {
+        /* clang-format off */
+        make_thread(1),
+        make_pid(1),
+        make_marker(TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+        make_timestamp(10),
+        make_marker(TRACE_MARKER_TYPE_CPU_ID, 1),
+        make_instr(1),
+        make_instr(2),
+        make_exit(1),
+        make_footer(),
+        /* clang-format on */
+    };
+    std::vector<scheduler_t::input_reader_t> readers;
+    readers.emplace_back(std::unique_ptr<mock_reader_t>(new mock_reader_t(memrefs)),
+                         std::unique_ptr<mock_reader_t>(new mock_reader_t()), 1);
+
+    std::vector<scheduler_t::range_t> regions;
+    regions.emplace_back(4, 0);
```

Comment:
Would the test continue to work for s/4/3? Since the 3rd instr is the first instr to not exist, it tests the boundary condition.

Reply:
Changed it to 3 (yes test still passes).